src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -75,12 +127,21 @@ RouterExecutorStart(QueryDesc *queryDesc, int eflags, Task *task)
 	/* ensure that the task is not NULL */
 	Assert(task != NULL);
 
-	/* disallow transactions and triggers during distributed modify commands */
+	/* disallow triggers during distributed modify commands */
```

Comment:
I… don't remember. This was ported over from `pg_shard`, which may have been written with the assumption that we'll call whatever's next in the hook chain, which might well cause triggers to fire. Sorry, can't remember precisely what the deal was.